test/modules/DirectMethodReceiver/src/DirectMethodReceiver.cs

```diff
@@ -0,0 +1,90 @@
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    class DirectMethodReceiver : IDisposable
+    {
+        Guid batchId;
+        IConfiguration configuration;
+        long directMethodCount = 1;
+        ILogger logger;
+        ModuleClient moduleClient;
+        TestResultReportingClient testResultReportingClient;
+
+        public DirectMethodReceiver(
+            ILogger logger,
+            IConfiguration configuration)
+        {
+            Preconditions.CheckNotNull(logger, nameof(logger));
+            Preconditions.CheckNotNull(configuration, nameof(configuration));
+
+            TestResultReportingClient testResultReportingClient = null;
+            Option<Uri> testReportCoordinatorUrl = Option.Maybe(configuration.GetValue<Uri>("testResultCoordinatorUrl"));
+            if (testReportCoordinatorUrl.HasValue)
+            {
+                testResultReportingClient = new TestResultReportingClient
+                {
+                    BaseUrl = testReportCoordinatorUrl.Expect(
+                        () => new ArgumentException("testReportCoordinatorUrl is empty"))
+                        .AbsoluteUri
+                };
+            }
+
+            this.logger = logger;
+            this.configuration = configuration;
+            this.testResultReportingClient = testResultReportingClient;
+            this.batchId = Guid.NewGuid();
+        }
+
+        public void Dispose() => this.moduleClient?.Dispose();
+
+        async Task<MethodResponse> HelloWorldMethodAsync(MethodRequest methodRequest, object userContext)
+        {
+            this.logger.LogInformation("Received direct method call.");
+            // Send the report to Test Result Coordinator
+            await this.ReportTestResult();
+            return new MethodResponse((int)HttpStatusCode.OK);
+        }
+
+        public async Task ReportTestResult()
+        {
+            DirectMethodTestResult testResult = null;
+            if (this.testResultReportingClient != null)
+            {
+                testResult = new DirectMethodTestResult(this.configuration.GetValue<string>("IOTEDGE_MODULEID") + ".receive", DateTime.UtcNow)
+                {
+                    TrackingId = Option.Maybe(this.configuration.GetValue<string>("trackingId"))
+                        .Expect(() => new ArgumentException("TrackingId is empty")),
+                    BatchId = this.batchId.ToString(),
+                    SequenceNumber = this.directMethodCount.ToString(),
+                    Result = HttpStatusCode.OK.ToString()
+                };
+                await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResult);
+            }
+
+            this.directMethodCount++;
```

Comment:
move this line into HelloWorldMethodAsync method

Reply:
good catch! It was there before then I move things around and that got left behind